sandbox/host/src/sandbox/wasmer_backend.rs

```diff
@@ -69,6 +69,39 @@ impl Backend {
     }
 }
 
+#[derive(Default)]
+pub struct Env {
+    gas: Option<wasmer::Global>,
+}
+
+// WARNING: intentionally to avoid cyclic refs
+impl Clone for Env {
+    fn clone(&self) -> Self {
+        Self {
+            gas: self.gas.clone().map(|mut global| {
+                global.into_weak_instance_ref();
+
+                global
+            }),
+        }
+    }
+}
+
+// copypaste
+pub const GLOBAL_NAME_GAS: &str = "gear_gas";
```

Comment:
This one have to be a global which is set by runtime. You can make additional  host function and call it each process queue or add additional argument to instantiation, and add global set in instantiation.